sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java

```diff
@@ -0,0 +1,195 @@
+                    })
+                    .subscribe(
+                        incomingRelationship -> client.deleteRelationship(incomingRelationship.getSourceId(), incomingRelationship.getRelationshipId())
+                            .subscribe(
+                                aVoid -> System.out.println("Found and deleted incoming relationship: " + incomingRelationship.getRelationshipId()),
+                                throwable -> System.err.println("Delete incoming relationship error: " + throwable)
+                            ));
+
+                try {
+                    // Verify that the list relationships and list incoming relationships async operations have completed.
+                    if (deleteRelationshipsSemaphore.tryAcquire(2, MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS)) {
+                        // Now the digital twin should be safe to delete
+
+                        // Call APIs to delete the twins.
+                        client.deleteDigitalTwin(twinId)
+                            .doOnSuccess(aVoid -> {
+                                System.out.println("Deleted digital twin: " + twinId);
+                                deleteTwinsSemaphore.release();
+                            })
+                            .doOnError(throwable -> {
+                                if (throwable instanceof ErrorResponseException && ((ErrorResponseException) throwable).getResponse().getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                                    deleteTwinsSemaphore.release();
+                                } else {
+                                    System.err.println("Could not delete digital twin " + twinId + " due to " + throwable);
+                                }
+                            })
+                            .subscribe();
+                    }
+                } catch (InterruptedException e) {
+                    throw new RuntimeException("Could not cleanup the pre-existing resources: ", e);
+                }
+            });
+
+        // Verify that a semaphore has been released for each delete async operation, signifying that the async call has completed successfully..
+        boolean created = deleteTwinsSemaphore.tryAcquire(twins.size(), MaxWaitTimeAsyncOperationsInSeconds, TimeUnit.SECONDS);
+        System.out.println("Twins deleted: " + created);
+    }
+
+    /**
+     * Creates all twins specified in the DTDL->DigitalTwins directory.
+     * @throws IOException If an I/O error is thrown when accessing the starting file.
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire permits on a semaphore.
+     */
+    public static void createTwins() throws IOException, InterruptedException {
+        System.out.println("CREATE DIGITAL TWINS");
+        Map<String, String> twins = FileHelper.loadAllFilesInPath(TwinsPath);
+        final Semaphore createTwinsSemaphore = new Semaphore(0);
```

Comment:
To understand why do we need semaphore here? Do we have some limitations for parallel operations?

Reply:
No, not really. The semaphore is to ensure that we do not exit before the async operation has completed. We start all async operations and then release the semaphore only once the async call has completed.
That way, the subsequent operations are executed after the previous call has completed -> similar to doing a await on a Task.